packages/x-date-pickers/src/internals/hooks/useViews.tsx

```diff
@@ -144,17 +149,37 @@ export function useViews<TValue, TView extends DateOrTimeView>({
   const nextView: TView | null = views[viewIndex + 1] ?? null;
 
   const handleChangeView = useEventCallback((newView: TView) => {
+    if (newView === view) {
+      return;
+    }
     setView(newView);
 
     if (onViewChange) {
       onViewChange(newView);
     }
   });
 
+  const handleFocusedViewChange = useEventCallback(
```

Comment:
This method has only been moved and `viewToFocus: TView | null` changed from `viewToFocus: TView`.